video/filter/vf_d3d11vpp.c

```diff
@@ -134,6 +149,152 @@ static void destroy_video_proc(struct mp_filter *vf)
     p->vp_enum = NULL;
 }
 
+// cacluate the render output, given the video size and the window size.
+static void get_render_size(int input_w, int input_h,
+                                int window_w, int window_h,
+                                int *out_w, int *out_h)
+{
+    // if input larger than window, then keep it as it is.
+    if (input_w > window_w || input_h > window_h) {
+        *out_w = input_w;
+        *out_h = input_h;
+    } else {
+        // else scale to window_w,window_h as much as possible
+        float aspect_ratio = (float)input_w / input_h;
+        *out_w = window_w;
+        *out_h = (int)(window_w / aspect_ratio);
+
+        // if the height is still larger than the window height after scaling,
+        // adjust the width based on the window height
+        if (*out_h > window_h) {
+            *out_h = window_h;
+            *out_w = (int)(window_h * aspect_ratio);
+        }
+    }
+}
+
+static void SetSuperResNvidia(struct mp_filter *vf)
+{
+    struct priv *p = vf->priv;
+    if(p->sr_state !=0){
+        return;
+    }
+
+    GUID kNvidiaPPEInterfaceGUID = {
+        0xd43ce1b3,
+        0x1f4b,
+        0x48ac,
+        {0xba, 0xee, 0xc3, 0xc2, 0x53, 0x75, 0xe6, 0xf7}};
+    unsigned int kStreamExtensionVersionV1 = 0x1;
+    unsigned int kStreamExtensionMethodSuperResolution = 0x2;
+
+    struct {
+        UINT version;
+        UINT method;
+        UINT enable; // 1 to enable, 0 to disable
+    } stream_extension_info = {kStreamExtensionVersionV1,
+                                kStreamExtensionMethodSuperResolution,
+                                1u};
+
+
+    HRESULT hr = ID3D11VideoContext_VideoProcessorSetStreamExtension(p->video_ctx,p->video_proc,0,&kNvidiaPPEInterfaceGUID,
+        sizeof(stream_extension_info), &stream_extension_info);
+
+    if (FAILED(hr)) {
+        p->sr_state = -1;
+        MP_ERR(vf, "Failed to enable Nvidia RTX Super Resolution. Error code: %lx\n", hr);
+    } else {
+        p->sr_state = 1;
+        MP_INFO(vf, "Nvidia RTX Super Resolution enabled\n");
+    }
+}
+
+static void SetSuperResIntel(struct mp_filter *vf)
+{
+    struct priv *p = vf->priv;
+    if (p->sr_state != 0){
+        return;
+    }
+    GUID GUID_INTEL_VPE_INTERFACE = {
+		  0xedd1d4b9,
+		  0x8659,
+		  0x4cbc,
+		  {0xa4, 0xd6, 0x98, 0x31, 0xa2, 0x16, 0x3a, 0xc3} };
+
+    enum : UINT {
```

Comment:
Move all enum/struct definitions to the top or at least outside function. Also fix compilation errors with gcc.

Reply:
updated